lib/adal-angular.js

```diff
@@ -175,16 +185,25 @@ if (typeof module !== 'undefined' && module.exports) {
                 var routeChangeHandler = function (e, nextRoute) {
                     if (nextRoute && nextRoute.$$route && isADLoginRequired(nextRoute.$$route, _adal.config)) {
                         if (!_oauthData.isAuthenticated) {
-                            _adal.info('Route change event for:' + $location.$$path);
+                            _adal.info('Route change event for:' + $location.$$url);
                             loginHandler();
                         }
                     }
                 };
 
-                var stateChangeHandler = function (e, nextRoute) {
-                    if (nextRoute && isADLoginRequired(nextRoute, _adal.config)) {
+                var stateChangeHandler = function (e, toState, toParams, fromState, fromParams){
```

Comment:
You will not in the documentation http://angular-ui.github.io/ui-router/site/#/api/ui.router.state.$state that the '$stateChangeStart' event supports all the parameters I added.